db/osqlcomm.c

```diff
@@ -6416,22 +6416,23 @@ int osql_process_packet(struct ireq *iq, unsigned long long rqid, uuid_t uuid,
         char *tablename;
 
         tablename = (char *)osqlcomm_usedb_type_get(&dt, p_buf, p_buf_end);
+        bdb_lock_tablename_read(thedb->bdb_env, tablename, trans);
```

Comment:
If we get the table lock here, do we get it again later in record.c?

Reply:
(i.e., it is great to do it  here, but I would prefer if this would be the only place:) )

Reply:
We get tablelock for each row - This is the correct place to get it. `record.c` is too late. Cleanup will be a different PR. Already have issue #330 to track. Will have to reconcile with how we support tag requests.

This is bare minimum required for correctness at the moment.